lib/basedriver/commands/find.js

```diff
@@ -341,34 +359,66 @@ helpers.getScreenshotForImageFind = async function (screenWidth, screenHeight) {
 
-  const screenAR = screenWidth / screenHeight;
-  const shotAR = shotWidth / shotHeight;
-
-  if (screenAR === shotAR) {
-    log.info('Screenshot aspect ratio matched screen aspect ratio');
-  } else {
-    log.warn(`When trying to find an element, determined that the screen ` +
-             `aspect ratio and screenshot aspect ratio are different. Screen ` +
-             `is ${screenWidth}x${screenHeight} whereas screenshot is ` +
-             `${shotWidth}x${shotHeight}.`);
-    shotWidth = shotWidth / (shotAR / screenAR);
-    log.warn(`Resizing screenshot to ${shotWidth}x${shotHeight} to match ` +
-             `screen aspect ratio so that image element coordinates have a ` +
-             `greater chance of being correct.`);
-    imgObj = imgObj.resize(shotWidth, shotHeight);
-  }
-
-  // now we know the aspect ratios match, but there might still be a scale
-  // mismatch, so just resize based on the screen dimensions
-  if (screenWidth !== shotWidth) {
+  // const screenAR = screenWidth / screenHeight;
+  // const shotAR = shotWidth / shotHeight;
+
+  // if (screenAR === shotAR) {
+  //   log.info('Screenshot aspect ratio matched screen aspect ratio');
+  // } else {
+  //   log.warn(`When trying to find an element, determined that the screen ` +
+  //            `aspect ratio and screenshot aspect ratio are different. Screen ` +
+  //            `is ${screenWidth}x${screenHeight} whereas screenshot is ` +
+  //            `${shotWidth}x${shotHeight}.`);
+  //   shotWidth = shotWidth / (shotAR / screenAR);
+  //   log.warn(`Resizing screenshot to ${shotWidth}x${shotHeight} to match ` +
+  //            `screen aspect ratio so that image element coordinates have a ` +
+  //            `greater chance of being correct.`);
+  //   imgObj = imgObj.resize(shotWidth, shotHeight);
+  // }
+
+  // Resize based on the screen dimensions only if both width and height are mismatched
+  // since except for that, it might be a situation which is different window rect and
+  // screenshot size like `@driver.window_rect #=>x=0, y=0, width=1080, height=1794` and
+  // `"deviceScreenSize"=>"1080x1920"`
+  let scale;
+  if (screenWidth !== shotWidth && screenHeight !== shotHeight) {
     log.info(`Scaling screenshot from ${shotWidth}x${shotHeight} to match ` +
              `screen at ${screenWidth}x${screenHeight}`);
     imgObj = imgObj.resize(screenWidth, screenHeight);
+
+    scale = {xScale: (1.0 * screenWidth) / shotWidth, yScale: (1.0 * screenHeight) / shotHeight};
   }
 
-  return (await imgObj.getBuffer(imageUtil.MIME_PNG)).toString('base64');
+  b64Screenshot = (await imgObj.getBuffer(imageUtil.MIME_PNG)).toString('base64');
+  return {b64Screenshot, scale};
 };
 
+/**
+ * Get a image that will be used for template maching.
+ * Returns scaled image if scale ratio is provided.
+ *
+ * @param {string} b64Template - base64-encoded image used as a template to be
+ * matched in the screenshot
+ * @param {?ScreenshotScale} scale - scale of screen
```

Comment:
it might be better for the readability if you provide the default value for the argument if it is optional (`scale=null`)